File: Arrays and Hashing/217. Contains Duplicate/contains_duplicate.cpp
// 1. Naiive Solution
// 2. Alternative Solution

// Naiive solution
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        auto s = std::set {};
        for (auto n : nums) {
            if (set.contains(n)) { return false; }
             s.insert(n)
        }
        return true;
    }
};

// Alternative solution
class Solution {
public:
    // Check size of array is larger than constructed set
    bool containsDuplicate(vector<int>& nums) {
        return nums.size() > std::unordered_set<int>(nums.begin(), nums.end()).size();
    }
};

File: Arrays and Hashing/217. Contains Duplicate/contains_duplicate.py
# 1. Naiive Solution
# 2. Alternative Solution

# 1. Naiive Solution
# class Solution:
#     def containsDuplicate(self, nums: List[int]) -> bool:
#         s = set()
#         for num in nums:
#             if num in s:
#                 return True
#             s.add(num)
#         return False
        
# 2. Alternative Solution
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        return len(nums) > len(set(nums))